packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -152,116 +150,58 @@ function tryInferInputContext(change: OptionalChangeset): "empty" | "filled" | u
-		} => {
-			const srcToDst = new RegisterMap<
-				[dst: RegisterId, target: "nodeTargeting" | "cellTargeting"]
-			>();
-			const dstToSrc = new RegisterMap<RegisterId>();
-			for (const [src, dst, target] of moves) {
-				srcToDst.set(src, [dst, target]);
-				dstToSrc.set(dst, src);
-			}
-			return { srcToDst, dstToSrc };
-		};
-
-		let earliestReservedDetachId: RegisterId | undefined;
-		let inputContext: "empty" | "filled" | undefined;
-
-		const childChangesByOriginalId = new RegisterMap<TaggedChange<NodeChangeset>[]>();
-		// TODO: It might be possible to compose moves in place rather than repeatedly copy.
-		// Additionally, working out a 'register allocation' strategy which enables frequent cancellation of noop moves
-		// for sandwich rebases would help with cloning if in-place proves too difficult
-		const current = getBidirectionalMaps([]);
-		for (const { change, revision } of changes) {
-			inputContext ??= tryInferInputContext(change);
-			const withIntention = (id: RegisterId): RegisterId => {
-				if (id === "self") {
-					return id;
-				}
-				const intention = id.revision ?? revision;
-				return { revision: intention, localId: id.localId };
-			};
-
-			const nextSrcToDst = new RegisterMap<
-				[dst: RegisterId, target: "nodeTargeting" | "cellTargeting"]
-			>();
-			const nextDstToSrc = new RegisterMap<RegisterId>();
-
-			if (change.reservedDetachId !== undefined && earliestReservedDetachId === undefined) {
-				earliestReservedDetachId = withIntention(change.reservedDetachId);
-			}
-
-			// Compose all the things that `change` moved.
-			for (const [unintentionedSrc, unintentionedDst, target] of change.moves) {
-				const src = withIntention(unintentionedSrc);
-				const dst = withIntention(unintentionedDst);
-				let originalSrc = current.dstToSrc.get(src);
-				let currentTarget: "cellTargeting" | "nodeTargeting" | undefined;
-				if (originalSrc !== undefined) {
-					const [dst2, existingTarget] =
-						current.srcToDst.get(originalSrc) ?? fail("expected backward mapping");
-					assert(
-						areEqualRegisterIds(dst2, src),
-						0x855 /* expected consistent backward mapping */,
-					);
-					currentTarget = existingTarget;
-				} else {
-					originalSrc = src;
-				}
-				nextSrcToDst.set(originalSrc, [
-					dst,
-					target === "cellTargeting" &&
-					(currentTarget === undefined || currentTarget === "cellTargeting")
-						? "cellTargeting"
-						: "nodeTargeting",
-				]);
-				nextDstToSrc.set(dst, originalSrc);
+		const inputContext = tryInferInputContext(change1) ?? tryInferInputContext(change2);
+
+		const earliestReservedDetachId: RegisterId | undefined =
+			withRevisionOrUndefined(change1.reservedDetachId, revision1) ??
+			withRevisionOrUndefined(change2.reservedDetachId, revision2);
+
+		const { srcToDst, dstToSrc } = getBidirectionalMaps(change1.moves, revision1);
+		for (const [src, dst, target] of change2.moves) {
+			const srcWithRevision = withRevision(src, revision2);
```

Comment:
`dstToSrc` doesn't get updated in this loop.

Reply:
That is intentional. Maybe I should rename it to `dstToSrc1` or something to make that clearer.

Reply:
I see. I'm not sure that `dstToSrc1` is very clear. A comment above the declaration to explain what each of `srcToDst` and `dstToSrc` are meant to contain would be fine I think.